Ingress/controllers/gce/backends.go

```diff
@@ -40,7 +40,17 @@ func portKey(port int64) string {
 }
 
 func beName(port int64) string {
-	return fmt.Sprintf("%v-%d", backendPrefix, port)
+	// TODO: Pipe the clusterName through, for now it saves code churn to just
+	// grab it globally, especially since we haven't decided how to handle
+	// namespace conflicts in the Ubernetes context.
+	if *clusterName == "" {
+		return fmt.Sprintf("%v-%d", backendPrefix, port)
+	}
+	taggedName := fmt.Sprintf("%v-%d%v%v", backendPrefix, port, clusterNameDelimiter, *clusterName)
+	if len(taggedName) > nameLenLimit {
+		taggedName = taggedName[:nameLenLimit]
+	}
+	return taggedName
```

Comment:
This is definitely better; at least we won't break on long clusters. My concern is that it won't be uniquely identifying for cluster names that share prefixes, given how short the limit is. We use shared name prefixes on a lot of e2e cluster names, and I imagine that's not uncommon practice. What do you think of using truncated clusterName + hash of clusterName? that would make it semi-human readable for short cluster names while still providing uniqueness for test/debugging purposes.